### PR TITLE
fix(stepfunctions-tasks): batch job - can not use task input as array size

### LINKED ISSUE #6922: [StepFunctions] RunBatchJob: Can't use task input to set arraySize
Trying to use the task input to set `arraySize` on `RunBatchJob` leads to a validation error `Invalid value of arraySize. The array size can be between 2 and 10,000.`

I'm reporting this as a bug instead of a feature request because I think that stepfunctions task constructs should accept values from the task input for all parameters, and `arraySize` is an essential parameter to dynamically control parallelism.

### Reproduction Steps
```python
sfnt.RunBatchJob(
    job_definition=...,
    job_name=...,
    job_queue=...,
    array_size=sfn.Data.number_at("$.arraySize")
)
```

### Error Log

`jsii.errors.JSIIError: Invalid value of arraySize. The array size can be between 2 and 10,000.`

### Environment

  - **CLI Version      : 1.30.0**
  - **Framework Version: 1.30.0**
  - **OS               : Windows 10**
  - **Language         : Python**

### Other

Root cause of the error is the validation done in https://github.com/aws/aws-cdk/blob/master/packages/%40aws-cdk/aws-stepfunctions-tasks/lib/run-batch-job.ts#L196

--- 

This is :bug: Bug Report

### PR DESCRIPTION
**Pull request by nija-at**
### Commit Message

Also, clean up some error messaging to be direct, terse and show the
user input.

fixes #6922
### End Commit Message

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

File: packages/@aws-cdk/aws-stepfunctions-tasks/lib/run-batch-job.ts [hunk 1]
```diff
@@ -191,37 +191,26 @@ export class RunBatchJob implements sfn.IStepFunctionsTask {
...
     // validate arraySize limits
-    if (
-      props.arraySize !== undefined &&
-      (props.arraySize < 2 || props.arraySize > 10000)
-    ) {
-      throw new Error(
-        `Invalid value of arraySize. The array size can be between 2 and 10,000.`
-      );
+    if (props.arraySize !== undefined && !Token.isUnresolved(props.arraySize) &&
+      (props.arraySize < 2 || props.arraySize > 10_000)) {
```
Comment (shivlaks) [inline packages/@aws-cdk/aws-stepfunctions-tasks/lib/run-batch-job.ts]: is `10_000` intended?

Comment (nija-at) [reply packages/@aws-cdk/aws-stepfunctions-tasks/lib/run-batch-job.ts]: Yes. https://www.typescriptlang.org/docs/handbook/release-notes/typescript-2-7.html#numeric-separators

Comment (rix0rrr) [reply packages/@aws-cdk/aws-stepfunctions-tasks/lib/run-batch-job.ts]: I believe there is a validation routine in `core`, specifically for people writing TypeScript constructs, which will do `!isUnresolved` for you. Let me look it up.

Comment (rix0rrr) [reply packages/@aws-cdk/aws-stepfunctions-tasks/lib/run-batch-job.ts]: https://github.com/aws/aws-cdk/blob/master/packages/@aws-cdk/core/lib/token.ts#L199

Comment (shivlaks) [reply packages/@aws-cdk/aws-stepfunctions-tasks/lib/run-batch-job.ts]: > Yes. https://www.typescriptlang.org/docs/handbook/release-notes/typescript-2-7.html#numeric-separators

neat

**Pull request merged**